### ISSUE TITLE
Add documentation how to setup dedicated SKF and documentation for Docker using IP or Hostname

### ISSUE THREAD

**Issue by hakanbilgin** [blabla1337/skf-flask#384]
Docker file obliges me to run SKF on https://localhost. I tried following commands and having errors.

I want to install SKF on an internal server of my company so that everybody can reach but I couldn't succeed.

docker -D run  -e "ORIGIN=10.58.12.22" -e "JWT_SECRET=88a9e1bc-2c64-4eb8-9a54-e69cbf7fbffc" -e "HTTPS=false" -d -p 10.58.12.22:8230:80 --name=security-knowledge-framework-poc --restart=always  blabla1337/skf-flask

docker -D run  -e "ORIGIN=10.58.12.22" -e "JWT_SECRET=88a9e1bc-2c64-4eb8-9a54-e69cbf7fbffc" -d -p 10.58.12.22:8230:443 --name=security-knowledge-framework-poc --restart=always  blabla1337/skf-flask

I startup SKF with the commands above and "http://10.58.12.22:8230/" and "https://10.58.12.22:8230/" gives me "Invalid Host Header" error.

How can I start the docker image on a server ip?

**Comment by blabla1337** [comment]
Hi @hakanbilgin,

It's more then that you need to do, we will rewrite this ticket to include some documentation to setup a dedicated / IP / hostname based and release it in the SKF repo.

We will describe the steps that we used ourselves for the https://demo.securityknowledgeframework.org

**Comment by hakanbilgin** [comment]
Thanks a lot. I will be following this github page for the documentation. It can be great if you can update this ticket when the documentation is ready. Thanks again.

**Comment by martinmarsicano** [comment]
Hi all. 
For a local run without Docker I grab the bash files located in the docker folder and modify them to run locally. I also have to modify the site-tls.con and site.conf because my nginx installation takes a different sintax (I don't know if it is because of the OS or some new version of nginx). 

If what I do is valid, I could give a hand with this issue. Maybe creating a folder as Docker, but named Local and putting the files I use with the local references.

**Comment by omerlh** [comment]
Any update on this? This is a really awesome project and I would like to run it using docker...

**Comment by hakanbilgin** [comment]
Any updates? Me and my company is impatient to use this tool as a dedicated installation.

**Comment by martinmarsicano** [comment]
There is a Local installation documentation, since #440 . Maybe we can close this issue, @blabla1337 ?

**Comment by omerlh** [comment]
Does this will work somehow with docker compose? Becuase in order for this to work on docker compose, it will need to listen on something different than localhost, and that what I'm looking for. 
Also, is there an option to use Let's Encrypt or something else to change the TLS certificate?

**Comment by omerlh** [comment]
Does this will work somehow with docker compose? Becuase in order for this to work on docker compose, it will need to listen on something different than localhost, and that what I'm looking for. 
Also, is there an option to use Let's Encrypt or something else to change the TLS certificate?

**Comment by Mason-Fan** [comment]
any update on this?

**Comment by uzzzval** [comment]
I am facing a similar issue. Any update on this ?

**Comment by hakan-bilgin** [comment]
Any updates for the installation on a dedicated server?

**Comment by aarvee11** [comment]
Easiest thing to do is to replace the Host Header with "localhost" if you are hosting it behind a load balancer that has L7 Request Rewrite Capabilities. For eg. in HAPROXY, you can use the rule as given below:

```
http-request replace-value Host skf.example.com localhost
```

**Comment by hakan-bilgin** [comment]
I think I found the solution. Problem is with skf-angular.sh file. I thınk hostname should be passed to ng serve command. See here: https://github.com/angular/angular-cli/issues/6349

I updated the skf-angular.sh with: 

**ng serve --configuration=production --host 0.0.0.0 --disable-host-check** 

and re-built the docker image and no more Invalid Header error.

I can now set up a dedicated skf with my docker image but still can't use port 80. Because all XHR calls of the application are hard coded as https.
